### PR TITLE
Test Summary UI Rewrite

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
Had to transform the Test Summary UI to fit the new design.

This would be a scenario when the candidate fails the test.
img_0010

This would be a scenario when the candidate passes the test.
img_0009

### CODE REVIEW

**Comment by johnmorleybjss** [comment]
Could you post a screenshot please?

**Comment by szabolcs-keresztes** [comment]
Sure, I'll post some screenshots very shortly.

File: src/components/test-result-header/test-result-header.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<ion-grid>
+    <ion-row>
+        <ion-col col-3 [ngClass]="summaryMetadata.selectedManoeuvre ? 'action-completed' : 'action-incomplete'">
+            <div [ngClass]="'test-section'">
+                <div [ngClass]="'test-section-text'">Manoeuvre</div>
```
Comment (johnmorleybjss) [inline src/components/test-result-header/test-result-header.html]: Perhaps it's better to do a standard `class` where the class name is fixed?

File: src/components/test-result-header/test-result-header.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            <div [ngClass]="'test-section'">
+                <div [ngClass]="'test-section-text'">ECO</div>
+            </div>
+        </ion-col>
+        <ion-col col-4>
+            <div [ngClass]="'test-section'">
+                ETA:
+                <span *ngFor="let action of etas; let i = index" [ngClass]="'eta-action'">
+                    {{action}}
+                    <span *ngIf="etas.length > 1 && i !== etas.length - 1" [ngClass]="eta-action">,</span>
```
Comment (johnmorleybjss) [inline src/components/test-result-header/test-result-header.html]: Might be an idea to refactor this predicate into a function in the controller class.

File: src/components/test-result-header/test-result-header.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  constructor() {}
+
+  ngOnInit() {
+    this.formatEtas();
+    this.formatEcoSelections();
+  }
+
+  formatEtas() {
+    this.etas = Object.keys(this.summaryMetadata.eta)
```
Comment (johnmorleybjss) [inline src/components/test-result-header/test-result-header.ts]: Neat 😄

File: src/components/test-summary/test-summary-section/test-summary-section.html [hunk 4]
```diff
@@ -1,10 +1,11 @@
-<ion-card class="mes-test-summary-section-card" [ngClass]="{'hidden': summary.total === 0}">
-    <ion-card-header margin-bottom [color]="getFaultTitleColour(summary.title)">
-        {{summary.title}} - {{summary.total}}
-    </ion-card-header>
-    <ion-card-content>
-        <ul class="mes-test-summary-list">
-            <li *ngFor="let fault of summary.faults">{{fault.name}} <span *ngIf="fault.total > 1">({{fault.total}})</span></li>
+<ion-row>
+    <ion-col col-4>
+        <div id="summary-total">{{summary.total}}</div>
```
Comment (johnmorleybjss) [inline src/components/test-summary/test-summary-section/test-summary-section.html]: Is this section component to be used multiple times on the same page? If so, `class` probably better than `id`. I'm not sure of the convention, but I don't see any per-instance ID prefixing in the DOM.

File: src/components/test-summary/test-summary.html [hunk 5]
```diff
@@ -1,50 +1,7 @@
...
-                </li>
-            </ul>
-        </ion-col>
-        <ion-col>
-            <test-summary-section [summary]="dangerousFaultSummary" *ngIf="dangerousFaultSummary.total > 0"></test-summary-section>
-            <test-summary-section [summary]="seriousFaultSummary" *ngIf="seriousFaultSummary.total > 0"></test-summary-section>
-            <test-summary-section [summary]="drivingFaultSummary" *ngIf="drivingFaultSummary.total > 0"></test-summary-section>
-        </ion-col>
-    </ion-row>
+    <test-summary-section [summary]="dangerousFaultSummary" *ngIf="dangerousFaultSummary.total > 0"></test-summary-section>
```
Comment (johnmorleybjss) [inline src/components/test-summary/test-summary.html]: You hero.

File: src/components/test-summary/test-summary.scss [hunk 6]
```diff
@@ -1,45 +1,3 @@
...
-            color: map-get($colors, mesPrimary);
-          }
-          &[name='close'] {
-            color: map-get($colors, failRed);
-          }
-        }
-      }
-    }
-  }
+  font-size: 20px;
```
Comment (johnmorleybjss) [inline src/components/test-summary/test-summary.scss]: See above.

File: src/pages/test-result/test-result.ts [hunk 7]
```diff
@@ -30,16 +30,29 @@ export class TestResultPage {
     this.faultStore
       .getFaultTotals()
       .subscribe((faultSummaries) => (this.faultSummaries = faultSummaries));
-    this.testResult = this.faultStore.getTestResult();
+    this.testResult =
+      this.faultStore.getTestResult() === TestResult.Fail ? 'unsuccessful' : 'passed';
     this.summaryMetadata = this.summaryMetadataService.getMetadata();
   }
 
+  getTestOutcomeContentClass() {
```
Comment (johnmorleybjss) [inline src/pages/test-result/test-result.ts]: I like it.

**Comment by szabolcs-keresztes** [comment]
Here is a screenshot when the test was successful

passed-test-summary

**Comment by szabolcs-keresztes** [comment]
Screenshot of a failed test summary, with numerous faults

failed-test-summary

File: src/components/test-result-header/test-result-header.html (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/components/test-result-header/test-result-header.html]: Okay, I can do that. Keep it consistent

**Comment by Cavalier07** [comment]
A couple of things I notice:

1) Your missing the header
2) There shouldn't be a space before the comma when you have multiple ETA's

File: src/pages/test-result/test-result.html [hunk 8]
```diff
@@ -4,22 +4,32 @@
...
+            <ion-card-header [ngClass]="card-header">
+                Faults recorded
+            </ion-card-header>
+            <ion-card-content>
+                <test-summary [faultSummaries]="faultSummaries" [testResult]="testResult" [summaryMetadata]="summaryMetadata"></test-summary>
+            </ion-card-content>
+        </div>
     </ion-card>
+    <div class="test-result-footer" text-center>
+        <button [navPush]="getNextPage()" class="mes-button-primary confirm-button">End debrief</button>
```
Comment (Cavalier07) [inline src/pages/test-result/test-result.html]: Are you able to pass the test slot data through to the next page(s)? You will need it for the header to get the candidate name on your page.

File: src/components/test-result-header/test-result-header.html (hunk 2 above)
Comment (szabolcs-keresztes) [reply src/components/test-result-header/test-result-header.html]: Extracted it

File: src/components/test-summary/test-summary.scss (hunk 6 above)
Comment (szabolcs-keresztes) [reply src/components/test-summary/test-summary.scss]: Sorry, what's that I should check above?

Comment (johnmorleybjss) [reply src/components/test-summary/test-summary.scss]: Was just me approving of lots of deleted code, you can safely ignore.

File: src/pages/test-result/test-result.html (hunk 8 above)
Comment (szabolcs-keresztes) [reply src/pages/test-result/test-result.html]: Sure thing, will do that!

File: src/components/test-summary/test-summary.scss (hunk 6 above)
Comment (szabolcs-keresztes) [reply src/components/test-summary/test-summary.scss]: Cool, cheers!

File: src/pages/test-result/test-result.html [hunk 9]
```diff
@@ -1,25 +1,35 @@
...
+            <ion-card-header class="card-header">
+                Faults recorded
+            </ion-card-header>
+            <ion-card-content>
+                <test-summary [faultSummaries]="faultSummaries" [testResult]="testResult" [summaryMetadata]="summaryMetadata"></test-summary>
+            </ion-card-content>
+        </div>
     </ion-card>
+    <div class="test-result-footer" text-center>
+        <button [navPush]="getNextPage()" class="mes-button-primary confirm-button">End debrief</button>
```
Comment (ammarhaiderbjss) [inline src/pages/test-result/test-result.html]: Could we use the new button style as per the [component library](https://wiki.i-env.net/display/MES/Component+library)
I think the class name is `.mes-cta-button-primary`

**Comment by szabolcs-keresztes** [comment]
@Cavalier07 Added the Header and removed the space before the comma on the ETA section.

**Pull request merged**